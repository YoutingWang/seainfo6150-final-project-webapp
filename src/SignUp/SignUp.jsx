import React, { useState } from "react";
import styles from "./SignUp.module.css"
import logo from "../images/Logo.jpg"
const SignUp = () => {
   const [submittedForm, setSubmittedForm] = useState();
  function onSubmit(e) {
    e.preventDefault();
    const data = new FormData(e.target);
    setSubmittedForm(data);
  };

  return (
    <div>
      <div className={styles.div}>
                <h2 className = {styles.text}>Sign up</h2>
                <img className={styles.img} src={logo} alt="logo"/>
            </div>
      {
        submittedForm ? (
          <div>Welcome {submittedForm.get("User Name")}. Enjoy!</div>
        ) : (
            <form className={styles.form} onSubmit={onSubmit}>
            <label htmlFor="name">User Name</label>
            <input className={styles.input} type="text" required = "requried" name = "User Name" placeholder = "Name" id = "name"/>
            <label htmlFor="email">Email</label>
            <input className={styles.input} type = "email" required = "required" name = "Email" placeholder = "Email" id = "email"/> 
            <label htmlFor="password">Password</label>
            <input className={styles.input} type = "password" required = "required" name = "Password" placeholder = "Password" id = "password"/>
            <label htmlFor="zipcode">Zipcode</label>
            <input className={styles.input} type="text" pattern="[0-9]{5}" required = "required" name = "Zipcode" placeholder = "Zipcode" id = "zipcode"/>
            <label htmlFor="phonenumber">Phone Number</label>
            <input className={styles.input} type = "tel" required = "required" name = "PhoneNumber" placeholder = "PhoneNumber" id = "phonenumber"/>
            
            <label htmlFor="gender">Gender</label><br/>
            <input type = "checkbox" id = "male"/>
            <label htmlFor="male">Male</label>
            <input type = "checkbox" id = "female"/>
            <label htmlFor="female">Female</label>
            <input type = "checkbox" id = "none"/>
            <label htmlFor="n/a">Decline to self identify</label><br/>

            <label htmlFor="Locations">Location</label>
            <select name="location" id="location">
              <option value="San Jose">San jose</option>
              <option value="LA">LA</option>
              <option value="New York">New York</option>
            </select><br/>
            <input className={styles.button} type="submit" value="Sign up" />
          </form>
        )
      }
    </div>
  )
}

export default SignUp