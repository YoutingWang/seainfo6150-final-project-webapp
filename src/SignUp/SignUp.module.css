.input{
    color: #1b1b1b;
    display: block;
    position: relative;
    font-size: 14px;
    margin: 17px;
    padding: 16px;
    text-align: center;
    font-family: 'Raleway', sans-serif;
}

.checkbox{
    color: #303030;
    position: relative;
    padding: 25px;
    font-family: 'Nunito', sans-serif;
    margin: 40px;
}

.button{
    color: black;
    text-align: center;
    margin: 24px;
    font-family: 'Nunito', sans-serif;
}

.form {
    width: 300px;
    height: 300px;
    margin-top: 10px;
    margin-left: auto;
    margin-right: auto;
}

.img{
    margin-top: 20px;
    height: 35px;
    width: 35px;
}

.div {
    margin-left: 45%;
    display: flex;
    flex-direction: row;
}

@media screen and (min-width: 550px) {
    .input{
        font-size: 14px;
        margin: 17px;
        padding: 16px;
    }
    .checkbox{
        padding: 25px;
        margin: 40px;
    }
    .form {
        width: 300px;
        height: 300px;
        margin-top: 4px;
    }
    .button{
        margin: 24px;
    }
}

@media screen and (min-width: 700px) {
    .input{
        font-size: 15px;
        margin: 19px;
        padding: 17px;
    }
    .checkbox{
        padding: 30px;
        margin: 45px;
    }
    .form {
        width: 320px;
        height: 320px;
        margin-top: 5px;
    }
    .button{
        margin: 26px;
    }
}

@media screen and (min-width: 900px) {
    .input{
        font-size: 20px;
        margin: 25px;
        padding: 22px;
    }
    .checkbox{
        padding: 30px;
        margin: 45px;
    }
    .form {
        width: 330px;
        height: 330px;
        margin-top: 7px;
    }
    .button{
        margin: 28px;
    }
}