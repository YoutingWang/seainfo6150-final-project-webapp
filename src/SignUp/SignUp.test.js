import React from "react";
import {
    render
} from "@testing-library/react";
import SignUp from "./SignUp.jsx";

it("renders the home component correctly", () => {
    const {
        container
    } = render( < SignUp / > );
    expect(container).toMatchSnapshot();
});