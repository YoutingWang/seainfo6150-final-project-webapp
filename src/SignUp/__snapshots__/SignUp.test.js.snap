// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the home component correctly 1`] = `
<div>
  <div>
    <div
      class="div"
    >
      <h2
        class="text"
      >
        Sign up
      </h2>
      <img
        alt="logo"
        class="img"
        src="Logo.jpg"
      />
    </div>
    <form
      class="form"
    >
      <label
        for="name"
      >
        User Name
      </label>
      <input
        class="input"
        id="name"
        name="User Name"
        placeholder="Name"
        required=""
        type="text"
      />
      <label
        for="email"
      >
        Email
      </label>
      <input
        class="input"
        id="email"
        name="Email"
        placeholder="Email"
        required=""
        type="email"
      />
      <label
        for="password"
      >
        Password
      </label>
      <input
        class="input"
        id="password"
        name="Password"
        placeholder="Password"
        required=""
        type="password"
      />
      <label
        for="zipcode"
      >
        Zipcode
      </label>
      <input
        class="input"
        id="zipcode"
        name="Zipcode"
        pattern="[0-9]{5}"
        placeholder="Zipcode"
        required=""
        type="text"
      />
      <label
        for="phonenumber"
      >
        Phone Number
      </label>
      <input
        class="input"
        id="phonenumber"
        name="PhoneNumber"
        placeholder="PhoneNumber"
        required=""
        type="tel"
      />
      <label
        for="gender"
      >
        Gender
      </label>
      <br />
      <input
        id="male"
        type="checkbox"
      />
      <label
        for="male"
      >
        Male
      </label>
      <input
        id="female"
        type="checkbox"
      />
      <label
        for="female"
      >
        Female
      </label>
      <input
        id="none"
        type="checkbox"
      />
      <label
        for="n/a"
      >
        Decline to self identify
      </label>
      <br />
      <label
        for="Locations"
      >
        Location
      </label>
      <select
        id="location"
        name="location"
      >
        <option
          value="San Jose"
        >
          San jose
        </option>
        <option
          value="LA"
        >
          LA
        </option>
        <option
          value="New York"
        >
          New York
        </option>
      </select>
      <br />
      <input
        class="button"
        type="submit"
        value="Sign up"
      />
    </form>
  </div>
</div>
`;
